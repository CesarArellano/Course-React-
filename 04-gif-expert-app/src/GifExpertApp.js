import React from 'react';

const GifExpertApp = () => {
  const categories = ['One Punch', 'Samurai X', 'Dragon Ball']
  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <ul>
        <li>One</li>
        <li>Samurai</li>
        <li>Dragon</li>
      </ul>
    </>
  );
}

export default GifExpertApp;