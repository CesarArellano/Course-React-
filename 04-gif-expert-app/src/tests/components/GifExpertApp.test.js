import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import GifExpertApp  from "../../GifExpertApp";

describe('Testing <GifExpertApp> Component', () => {
  test('Must display correctly this component.', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Must show a categories list', () => {
    const categories = ['One Punch', 'Dragon Ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });  
  
});
