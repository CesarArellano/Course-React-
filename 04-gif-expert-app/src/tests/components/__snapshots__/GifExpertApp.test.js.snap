// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifExpertApp> Component Must display correctly this component. 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <button
    onClick={[Function]}
  >
    Agregar
  </button>
</Fragment>
`;
